src/rgw/rgw_common.h

```diff
@@ -690,6 +694,43 @@ void decode_json_obj(rgw_placement_rule& v, JSONObj *obj);
 inline std::ostream& operator<<(std::ostream& out, const rgw_placement_rule& rule) {
   return out << rule.to_str();
 }
+
+class RateLimiter;
+struct RGWQoSInfo {
```

Comment:
maybe we should move this into its own file?
why do we add more stuff to rgw_common?

Reply:
https://github.com/ceph/ceph/pull/42891#discussion_r694915814